src/main/fc/fc_core.c

```diff
@@ -194,115 +194,122 @@ static void updateArmingStatus(void)
-            ENABLE_ARMING_FLAG(ARMING_DISABLED_NAVIGATION_UNSAFE);
+            SET_ARMING_DISABLED_FLAG(ARMING_DISABLED_NAVIGATION_UNSAFE);
         }
         else {
-            DISABLE_ARMING_FLAG(ARMING_DISABLED_NAVIGATION_UNSAFE);
+            UNSET_ARMING_DISABLED_FLAG(ARMING_DISABLED_NAVIGATION_UNSAFE);
         }
 #endif
 
 #if defined(USE_MAG)
         /* CHECK: */
         if (sensors(SENSOR_MAG) && !STATE(COMPASS_CALIBRATED)) {
-            ENABLE_ARMING_FLAG(ARMING_DISABLED_COMPASS_NOT_CALIBRATED);
+            SET_ARMING_DISABLED_FLAG(ARMING_DISABLED_COMPASS_NOT_CALIBRATED);
         }
         else {
-            DISABLE_ARMING_FLAG(ARMING_DISABLED_COMPASS_NOT_CALIBRATED);
+            UNSET_ARMING_DISABLED_FLAG(ARMING_DISABLED_COMPASS_NOT_CALIBRATED);
         }
 #endif
 
         /* CHECK: */
         if (sensors(SENSOR_ACC) && !STATE(ACCELEROMETER_CALIBRATED)) {
-            ENABLE_ARMING_FLAG(ARMING_DISABLED_ACCELEROMETER_NOT_CALIBRATED);
+            SET_ARMING_DISABLED_FLAG(ARMING_DISABLED_ACCELEROMETER_NOT_CALIBRATED);
         }
         else {
-            DISABLE_ARMING_FLAG(ARMING_DISABLED_ACCELEROMETER_NOT_CALIBRATED);
+            UNSET_ARMING_DISABLED_FLAG(ARMING_DISABLED_ACCELEROMETER_NOT_CALIBRATED);
         }
 
         /* CHECK: */
         if (!isHardwareHealthy()) {
-            ENABLE_ARMING_FLAG(ARMING_DISABLED_HARDWARE_FAILURE);
+            SET_ARMING_DISABLED_FLAG(ARMING_DISABLED_HARDWARE_FAILURE);
         }
         else {
-            DISABLE_ARMING_FLAG(ARMING_DISABLED_HARDWARE_FAILURE);
+            UNSET_ARMING_DISABLED_FLAG(ARMING_DISABLED_HARDWARE_FAILURE);
         }
 
         /* CHECK: BOXFAILSAFE */
         if (IS_RC_MODE_ACTIVE(BOXFAILSAFE)) {
-            ENABLE_ARMING_FLAG(ARMING_DISABLED_BOXFAILSAFE);
+            SET_ARMING_DISABLED_FLAG(ARMING_DISABLED_BOXFAILSAFE);
         }
         else {
-            DISABLE_ARMING_FLAG(ARMING_DISABLED_BOXFAILSAFE);
+            UNSET_ARMING_DISABLED_FLAG(ARMING_DISABLED_BOXFAILSAFE);
         }
 
         /* CHECK: BOXKILLSWITCH */
         if (IS_RC_MODE_ACTIVE(BOXKILLSWITCH)) {
-            ENABLE_ARMING_FLAG(ARMING_DISABLED_BOXKILLSWITCH);
+            SET_ARMING_DISABLED_FLAG(ARMING_DISABLED_BOXKILLSWITCH);
         }
         else {
-            DISABLE_ARMING_FLAG(ARMING_DISABLED_BOXKILLSWITCH);
+            UNSET_ARMING_DISABLED_FLAG(ARMING_DISABLED_BOXKILLSWITCH);
         }
 
         /* CHECK: Do not allow arming if Servo AutoTrim is enabled */
         if (IS_RC_MODE_ACTIVE(BOXAUTOTRIM)) {
-	       ENABLE_ARMING_FLAG(ARMING_DISABLED_SERVO_AUTOTRIM);
-	    }
-        else {
-	       DISABLE_ARMING_FLAG(ARMING_DISABLED_SERVO_AUTOTRIM);
-	    }
+            SET_ARMING_DISABLED_FLAG(ARMING_DISABLED_SERVO_AUTOTRIM);
+        } else {
+            UNSET_ARMING_DISABLED_FLAG(ARMING_DISABLED_SERVO_AUTOTRIM);
+        }
+
+        if (isModeActivationConditionPresent(BOXPREARM)) {
```

Comment:
`isModeActivationConditionPresent` is expensive in terms of CPU time. We should process it once and expose it as a special function call reading a boolean flag, similar to checking for navigation modes.